File: api/src/main/java/org/delivery/api/domain/store/business/StoreBusiness.java
package org.delivery.api.domain.store.business;

import lombok.RequiredArgsConstructor;
import org.delivery.api.common.annotation.Business;
import org.delivery.api.domain.store.controller.model.StoreRegisterRequest;
import org.delivery.api.domain.store.controller.model.StoreResponse;
import org.delivery.api.domain.store.converter.StoreConverter;
import org.delivery.api.domain.store.service.StoreService;
import org.delivery.db.store.enums.StoreCategory;

import java.util.List;
import java.util.stream.Collectors;

@RequiredArgsConstructor
@Business
public class StoreBusiness {
    private final StoreService storeService;
    private final StoreConverter storeConverter;


    public StoreResponse register(
            StoreRegisterRequest storeRegisterRequest
    ){
        var entity = storeConverter.toEntity(storeRegisterRequest);
        var newEntity = storeService.register(entity);
        return storeConverter.toResponse(newEntity);
    }

    //조회
    public List<StoreResponse> searchCategory(
            StoreCategory category
    ){

        var storeList = storeService.searchByCategory(category);
        return storeList.stream()
                .map(storeConverter::toResponse)
                .collect(Collectors.toList());
    }
}

File: api/src/main/kotlin/org/delivery/api/domain/store/business/StoreBusiness.kt
package org.delivery.api.domain.store.business

import org.delivery.api.domain.store.controller.model.StoreRegisterRequest
import org.delivery.api.domain.store.controller.model.StoreResponse
import org.delivery.api.domain.store.converter.StoreConverter
import org.delivery.api.domain.store.service.StoreService
import org.delivery.common.annotation.Business
import org.delivery.db.store.enums.StoreCategory

@Business
class StoreBusiness (
    val storeService:StoreService,
    val storeConverter: StoreConverter
){
    fun register(
        storeRegisterRequest: StoreRegisterRequest?
    ):StoreResponse{
        storeConverter.toEntity(storeRegisterRequest)
            .let{
                storeService.register(it)
            }.let{
                return storeConverter.toResponse(it)
            }
    }


    fun searchCategory(
        category:StoreCategory
    ):List<StoreResponse>{
        return storeService.searchByCategory(category)
            .map{
                storeConverter.toResponse(it)
            }.toList()
    }
}